### PR TITLE
Support dynamic parameters in Jet SQL engine

### PR DESCRIPTION
**Pull request by gierlachg**

### CODE REVIEW

**Review by alex-dukhno — APPROVED** [review]

**Review by viliam-durina — CHANGES_REQUESTED** [review]

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlan.java [hunk 1]
```diff
@@ -95,7 +96,8 @@ public boolean producesRows() {
         }
 
         @Override
-        public SqlResult execute(QueryId queryId) {
+        public SqlResult execute(QueryId queryId, List<Object> arguments) {
+            assert arguments.isEmpty();
```
[the remark this answers is not in the document]
Comment (gierlachg) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlan.java:100]: We should never reach this point for dynamic parameters and i.e. `CREATE MAPPING` - it fails at parsing. Also I don't see much value in creating tests for such cases as we would pretty much test `Calcite` parser.

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlanExecutor.java [hunk 2]
```diff
@@ -210,4 +216,30 @@ SqlResult execute(SelectOrSinkPlan plan, QueryId queryId) {
...
     }
+
+    private List<Object> prepareArguments(QueryParameterMetadata parameterMetadata, List<Object> arguments) {
+        assert arguments != null;
+
+        int parameterCount = parameterMetadata.getParameterCount();
+        if (parameterCount != arguments.size()) {
+            throw QueryException.error(
+                    SqlErrorCode.DATA_EXCEPTION,
+                    "Unexpected parameter count: expected " + parameterCount + ", got " + arguments.size()
```
[the remark this answers is not in the document]
Comment (gierlachg) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlanExecutor.java:231]: Yes. Added test.

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/UnknownStatistic.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ private UnknownStatistic() {
 
     @Override
     public Double getRowCount() {
-        return null;
+        return 0D;
```
[the remark this answers is not in the document]
Comment (gierlachg) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/UnknownStatistic.java]: Reverted back to `null` - had to change `HazelcastTable` a bit to accommodate that.

File: hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlCsvTest.java [hunk 4]
```diff
@@ -222,6 +242,23 @@ public void test_tableFunction() {
         );
     }
 
+    @Test
+    public void test_tableFunctionWithDynamicParameters() {
+        assertRowsAnyOrder(
+                "SELECT CAST(long AS BIGINT) - ?"
+                        + " FROM TABLE ("
+                        + "csv_file ("
+                        + " path => '" + RESOURCES_PATH + "'"
```
[the remark this answers is not in the document]
Comment (gierlachg) [reply hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlCsvTest.java:251]: The problem is we don't have access to arguments in file table functions. I can try to change that in subsequent PR but I would leave this as it is in this one.

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlan.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (viliam-durina) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlan.java:100]: I tried this code:
```java
sqlService.execute("create mapping t type TestBatch", "param0");
```
and got:
```
java.lang.AssertionError
	at com.hazelcast.jet.sql.impl.JetPlan$CreateMappingPlan.execute(JetPlan.java:100)
        ...
```

**Review by viliam-durina — APPROVED** [review]

**Pull request merged**